src/components/NcRichText/NcReferenceList.vue

```diff
@@ -83,7 +83,11 @@ export default {
 			}
 		},
 		fullUrl() {
-			return new URL(this.text.trim(), window.location)
+			const matchArray = this.text.trim().match(new RegExp(URL_PATTERN))
```

Comment:
Unfortunately this will not work as `URL_PATTERN` matches only full URLs, not relative or absolute URLs without a location.

Reply:
In fact the whole idea of #5272 was to turn relative links (e.g. `../Target`) and absolute links (e.g. `/index.php/apps/collectives/Garden/Watering`) into full URLs with a location *before*  they're matched against `URL_PATTERN` :grimacing: 

Could you provide an example text of what is passed to `NcReferenceList` as property `text` which breaks currently?

Reply:
`test message` - it tries to resolve a link: `https://nextcloud.local/index.php/call/test%20message`
`test message with a link https://nextcloud.local/index.php/call/w38mo5vy` - same: `https://nextcloud.local/index.php/call/test%20message%20with%20a%20link%20https://nextcloud.local/index.php/call/w38mo5vy`
`https://nextcloud.local/index.php/call/w38mo5vy` - resolves correctly

Reply:
Maybe we could replace URL_PATTERN with some combination RELATIVE_URL_PATTERN?

Reply:
Would it be an option to sanitize the links inside Talk and make sure to only pass valid URLs to `NcReferenceList`?

Given that we cannot distinguish between valid relative links and mere text I only see two paths forward:

* Either make sure only links (be it full URLs, relative or absolute paths) are passed to NcReferenceList
* Or move the logic to transform relative/absolute paths into full URLs to Text. That would mean that relative/absolute paths from other apps no longer get regarded by NcReferenceList.

Reply:
> Maybe we could replace URL_PATTERN with some combination RELATIVE_URL_PATTERN?

That could work if use something like `^(\.){1,2}\/.*` as regex for `RELATIVE_URL_PATTERN`.

Reply:
Did you mean one of these places?
https://github.com/nextcloud/text/blob/35be559ca27f384cd80e9e5fe76ee739c7c754e6/src/nodes/ParagraphView.vue#L105-L112
https://github.com/nextcloud/text/blob/35be559ca27f384cd80e9e5fe76ee739c7c754e6/src/components/Link/LinkBubbleView.vue#L148-L155

Reply:
Yeah, both probably. I'm fine with trying to handle this in Text. So let's revert my earlier PR and I'll look into it in Text.